telemetry/otel.go

```diff
@@ -0,0 +1,234 @@
+		propagation.Baggage{},
+	)
+}
+
+func newOTLPTraceProvider(ctx context.Context, res *resource.Resource) (*trace.TracerProvider, error) {
+	otelAgentAddr, ok := os.LookupEnv("OTEL_EXPORTER_OTLP_ENDPOINT")
+	if !ok {
+		otelAgentAddr = defaultOtelAgentAddr
+	}
+	traceClient := otlptracegrpc.NewClient(
+		otlptracegrpc.WithInsecure(),
+		otlptracegrpc.WithEndpoint(otelAgentAddr))
+	traceExporter, err := otlptrace.New(ctx, traceClient)
+	if err != nil {
+		return nil, err //nolint:wrapcheck // Want passthrough
+	}
+
+	bsp := trace.NewBatchSpanProcessor(traceExporter)
+	traceProvider := trace.NewTracerProvider(
+		trace.WithSampler(trace.AlwaysSample()),
+		trace.WithResource(res),
+		trace.WithSpanProcessor(bsp),
+	)
+	return traceProvider, nil
+}
+
+func newStdoutTraceProvider(_ context.Context, res *resource.Resource) (*trace.TracerProvider, error) {
+	traceExporter, err := stdouttrace.New(
+		stdouttrace.WithPrettyPrint())
+	if err != nil {
+		return nil, err //nolint:wrapcheck // Want passthrough
+	}
+
+	traceProvider := trace.NewTracerProvider(
+		trace.WithResource(res),
+		trace.WithBatcher(traceExporter,
+			trace.WithBatchTimeout(5*time.Second)),
+	)
+	return traceProvider, nil
+}
+
+func newOTLPMeterProvider(ctx context.Context, res *resource.Resource) (*metric.MeterProvider, error) {
+	otelAgentAddr, ok := os.LookupEnv("OTEL_EXPORTER_OTLP_ENDPOINT")
+	if !ok {
+		otelAgentAddr = defaultOtelAgentAddr
+	}
+	metricExporter, err := otlpmetricgrpc.New(
+		ctx,
+		otlpmetricgrpc.WithInsecure(),
+		otlpmetricgrpc.WithEndpoint(otelAgentAddr),
+		otlpmetricgrpc.WithTemporalitySelector(deltaSelector))
+	if err != nil {
+		return nil, err //nolint:wrapcheck // Want passthrough
+	}
+
+	meterProvider := metric.NewMeterProvider(
+		metric.WithResource(res),
+		metric.WithReader(metric.NewPeriodicReader(metricExporter,
+			metric.WithInterval(1*time.Minute))),
+	)
+	return meterProvider, nil
+}
+
+func newStdoutMeterProvider(_ context.Context, res *resource.Resource) (*metric.MeterProvider, error) {
+	metricExporter, err := stdoutmetric.New(
+		stdoutmetric.WithPrettyPrint(),
+		stdoutmetric.WithTemporalitySelector(deltaSelector))
+	if err != nil {
+		return nil, err //nolint:wrapcheck // Want passthrough
+	}
+
+	meterProvider := metric.NewMeterProvider(
+		metric.WithResource(res),
+		metric.WithReader(metric.NewPeriodicReader(metricExporter,
+			metric.WithInterval(1*time.Minute))),
+	)
+	return meterProvider, nil
+}
+
+// Borrowed from https://docs.datadoghq.com/opentelemetry/guide/otlp_delta_temporality/#configuring-your-opentelemetry-sdk
+func deltaSelector(kind metric.InstrumentKind) metricdata.Temporality {
+	switch kind {
+	case metric.InstrumentKindCounter, metric.InstrumentKindGauge, metric.InstrumentKindHistogram, metric.InstrumentKindObservableGauge, metric.InstrumentKindObservableCounter:
+		return metricdata.DeltaTemporality
+	case metric.InstrumentKindUpDownCounter, metric.InstrumentKindObservableUpDownCounter:
```

Comment:
Intentionally blank?

Reply:
yea the linter requires me to do it this way

https://github.com/TapestryEnergy/gridaware-backend/pull/320/commits/708d21a22e720792cd5a7955fc112aac39c9c5f8

Reply:
Don't have access to that repo

Reply:
Replied earlier.